### PR TITLE
Add local and store version infos

### PR DESCRIPTION
**Pull request by MysticFragilist**
Hi there!

First of, thanks for this package it's really nice! 😄 

I was about to use your package on our project but I was missing some things. In the project I'm working on, we needed one pretty specific way to handle update pop up. The project needs to popup the prompt alert only when a minor change is posed, but on a bugfix, we needed to force update the user.

So basically we wanted to implement our own custom `updateIsAvailable` with differents states. So what I would want for this package is to have a way to go fetch the store version the same way it's done internally. Also, for the sack of consistency, I added a `localVersion` getter to access the current downloaded version.

The only thing missing from this PR is maybe an actual test as I'm not sure how to test a dart package like this in an app properly. Don't know if you have something to test against on your side? Just to make sure the feature works.

Hopes my addition are up to standard, and again a big thanks for this package 😄

Don't hesitate to contact me for any questions on this PR,
Sam

### CODE REVIEW

File: example/main.dart [hunk 1]
```diff
@@ -33,6 +33,12 @@ class MyApp extends StatelessWidget {
                     child: const Text('Check Update'),
                     onPressed: () => siren.promptUpdate(context),
                   )
```
Comment (diegocosta) [inline example/main.dart:35]:
```suggestion
                  ),
```
We have a typo here, a comma is missing.

File: example/main.dart [hunk 2]
```diff
@@ -33,6 +33,12 @@ class MyApp extends StatelessWidget {
...
+                  Text(
+                    'Local Version: ${siren.localVersion}',
```
Comment (diegocosta) [inline example/main.dart:37]: This example isn't working so well, what do you think about rewriting using FutureBuilder or other ways to work with Future on Flutter?

Simulator Screen Shot - iPhone 13 - 2021-10-30 at 22 07 59

File: lib/src/siren.dart [hunk 3]
```diff
@@ -27,6 +27,21 @@ class Siren {
...
+
+  /// This method will get the store version of your app.
+  Future<Version> get localVersion async {
+    final packageInfo = await PackageInfo.fromPlatform();
+    final currentVersion = packageInfo.version;
+    return Version.parse(currentVersion);
+  }
+
   /// This method checks for an update in the application store and returns if there is a newer version than the local one.
   Future<bool> updateIsAvailable() async {
```
Comment (diegocosta) [inline lib/src/siren.dart:46]: Would be interesting to rewrite the `updateIsAvailable` method to use the getters to avoid code duplication.

**Review by diegocosta — CHANGES_REQUESTED** [review]
Hello there, first of all, thanks for your contribution, I'm really happy to hear about new ways to use this package! Let's work together in this implementation. 

LMK what do you think of the suggestions below.

File: example/main.dart (hunk 2 above)
Comment (MysticFragilist) [reply example/main.dart:37]: Oups my bad :/ I'll be on that 😄

**Comment by MysticFragilist** [comment]
Let me know if my changes work 😄

**Comment by diegocosta** [comment]
Hey @MysticFragilist, I found some issues with the example file, see:

Screen Shot 2021-10-31 at 03 14 47

So, I created a complete example project on the branch `working_example` to test your code there. Did you do a pull in your local branch, adapt your project moving your `example/main.dart` to `example/lib/main.dart` and fixes these issues?

**Comment by MysticFragilist** [comment]
Thanks! This will be really helpful to test with :smile: I'll update you once it's done 🚀

**Comment by MysticFragilist** [comment]
Hi @diegocosta! I think I found the issue I was having (did not push it tho). 
But, I found an issue with the package relating to my app. When using the package id from my app, `ca.etsmtl.applets.etsmobile` the only version it gets out is a dot "."
image

You can add this test to check out the error google_play_store_service_test.dart:
```dart
  test('test ca.etsmtl.applets.etsmobile', () async {
    final details = await SirenGooglePlayStore()
        .getStoreResponse(from: 'ca.etsmtl.applets.etsmobile');

    expect(details.version, '4.4.8');
    expect(details.package, 'ca.etsmtl.applets.etsmobile');
  });
```

**Comment by MysticFragilist** [comment]
Okay right now, by trial and error, I manage to make it work by adding a `<\/div></span>` at the end of the regex. It seems like the lookahead expression finds multiple `</span>` values and don't return the right one for my app. (The weird thing is that was working on all the other app I tested it out with). Right now, it's working for my app and the other one I tested them with.

**Comment by diegocosta** [comment]
@MysticFragilist This is really weird! The source code is no different for your package. So, your suggestion works, you can send in this PR.  Nice catch!

**Comment by diegocosta** [comment]
@MysticFragilist I'll approve your PR now!

**Review by diegocosta — APPROVED** [review]

**Comment by MysticFragilist** [comment]
> @MysticFragilist This is really weird! The source code is no different for your package. So, your suggestion works, you can send in this PR. Nice catch!

Yes, It was really weird I copied all the HTML in a regex parser and it wasn't giving any different output than any other page from google play. Regex is so obscure sometimes hahaha, but eh if it works now why not! 😄 

Thanks for all the help on this, take care! 💯

**Comment by diegocosta** [comment]
@MysticFragilist Your brilliant work was published on pub.dev in version 1.2.0

**Pull request merged**